### ISSUE TITLE
How to handle file type fields ?

### ISSUE THREAD

**Issue by guillbdx** [Textalk/angular-schema-form#271]
Hi !

Congrats for your work. It looks like there is neither native nor add-on form field type for handling file type fields. Do you have any recommandation about this issue ?

Thanks.

**Comment by Dervisevic** [comment]
So, it has been brought up before, for example #160. It has been agreed upon that it won't be native but an add-on would be excellent. I think i've seen people working on them maybe even have them but not published. The only concrete example i found is in #87 and it would seem @israelidanny may have something.

**Comment by gpedelagrabe** [comment]
Hi, thanks. Your links were useful, in particular this one : https://github.com/mike-marcacci/gandhi/tree/master/lib/modules/gandhi-decorator-upload

I started today an implementation using [angular-file-upload](https://github.com/danialfarid/angular-file-upload). Seems to work fine, even though there's still a lot do : 
https://github.com/Webfutur/angular-json-form-file-type

**Comment by binnyg** [comment]
I tried a very simple implementation of file upload(https://gist.github.com/binnyg/bc7df2858c13b7c7c10a). File upload works fine but I am having trouble with validation. I defined the property as an object but when I upload I get the following error - **invalid type: array (expected object)**. This error makes sense so I changed the type to array but now the form is not rendered. 

Need some advice and help to fix the issue. Thanks.
